src/sentry/api/endpoints/group_similar_issues_embeddings.py

```diff
@@ -32,16 +30,26 @@ def get_value_if_exists(exception_value):
 
 
 def get_stacktrace_string(data):
-    # Get the in-app data
     if (
         not data.get("app")
+        or not data["app"].get("hash")
         or not data["app"].get("component")
         or not data["app"]["component"].get("values")
+    ) and (
+        not data.get("system")
+        or not data["system"].get("hash")
+        or not data["system"].get("component")
+        or not data["system"]["component"].get("values")
```

Comment:
consider using our util function `safe_path` to simplify this https://github.com/getsentry/sentry/blob/773e7d859f480d8be3730fe2be31f6fb353f57c4/src/sentry/utils/safe.py#L108

additionally, with conditionals like this, consider using De Morgan's law to simplify: https://en.wikipedia.org/wiki/De_Morgan%27s_laws

Not A or Not B or Not C => Not (A and B and C)

Reply:
thank you for this suggestion! super helpful